mypy/meet.py

```diff
@@ -192,10 +191,10 @@ def is_overlapping_tuples(t: Type, s: Type, use_promotions: bool) -> Optional[bo
                 if all(is_overlapping_types(ti, si, use_promotions)
                        for ti, si in zip(t.items, s.items)):
                     return True
-        # TupleType and non-tuples do not overlap
-        return False
-    # No tuples are involved here
-    return None
+        return is_subtype(t, s) or is_subtype(s, t)
```

Comment:
> Anyway, the selfcheck test seems to perform the same within a few hundredths of a second on master and this branch (without cache in both cases), so I am not concerned about performance regressions.

Did you measure performance when this check is here or at the top of the `is_overlapping_types`? If at the top of `is_overlapping_types`, then I would rather add `if is_subtype(t, s) or is_subtype(s, t): return True` at the top of `is_overlapping_types` (after `builtins.object` check) and leave this function completely "intact" then.

Reply:
It was indeed measured from the top. I just added it after the object check.